darts/tests/models/forecasting/test_torch_forecasting_model.py

```diff
@@ -2515,6 +2517,156 @@ def test_predict_after_fit_from_dataset(self):
+            )
+
+            # for best model
+            checkpoints_folder_best = os.path.join(tmpdir_fn, "best_checkpoints")
+            checkpoint_callback_best = ModelCheckpoint(
+                dirpath=checkpoints_folder_best,
+                save_last=True,
+                monitor="val_loss",
+                filename="best-{epoch}-{val_loss:.2f}",
+            )
+            trainer_best = pl.Trainer(
+                max_epochs=n_epochs,
+                callbacks=[checkpoint_callback_best],
+                **tfm_kwargs["pl_trainer_kwargs"],
+            )
+
+        model_last = RNNModel(**kwargs_last)
+        model_last.fit(
+            series_train,
+            val_series=series_val,
+            load_best=False,
+            trainer=trainer_last,
+        )
+        last_model_epochs_trained = model_last.epochs_trained
+
+        model_best = RNNModel(**kwargs_best)
+        model_best.fit(
+            series_train, val_series=series_val, load_best=True, trainer=trainer_best
+        )
+        best_model_epochs_trained = model_best.epochs_trained
+
+        # The model trained to the end should have n_epochs trained
+        assert last_model_epochs_trained == n_epochs
+        # The model loading the best checkpoint should have trained fewer epochs (or equal if best is last)
+        # With our crafted data, it's highly likely to be less.
+        assert best_model_epochs_trained < last_model_epochs_trained
+
+        # We can also check that the trainer object confirms the best model was loaded
+        trainer_to_check = trainer_best if use_custom_trainer else model_best.trainer
+        best_model_path = trainer_to_check.checkpoint_callback.best_model_path
+        assert best_model_path and os.path.exists(best_model_path)
+
+        # predictions must work and be different
+        preds_last = model_last.predict(n=1)
+        preds_best = model_best.predict(n=1)
+        assert preds_last != preds_best
+
+    @pytest.mark.parametrize(
+        "save_checkpoints, val_series_provided", [(False, True), (True, False)]
+    )
+    def test_load_best_ignored(
+        self, tmpdir_fn, caplog, save_checkpoints, val_series_provided
+    ):
+        """Tests that `load_best` is ignored with a warning when conditions are not met."""
+        series_train = tg.sine_timeseries(length=50).astype(np.float32)
+        series_val = (
+            tg.sine_timeseries(length=25).astype(np.float32)
+            if val_series_provided
+            else None
+        )
+        n_epochs = 3
+
+        model_kwargs = {
+            "input_chunk_length": 12,
+            "model": "RNN",
+            "hidden_dim": 10,
+            "n_rnn_layers": 1,
+            "n_epochs": n_epochs,
+            "work_dir": tmpdir_fn,
+            "random_state": 42,
+            "save_checkpoints": save_checkpoints,
+            "model_name": f"test_load_best_ignored_{save_checkpoints}_{val_series_provided}",
```

Comment:
Let's reduce it to what is required for the test

```suggestion
            "n_epochs": n_epochs,
            "work_dir": tmpdir_fn,
            "random_state": 42,
            "save_checkpoints": save_checkpoints,
```